vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -77,8 +67,14 @@ def _get_callback_mapping(self):
         from vizro.actions._callback_mapping._get_action_callback_mapping import _get_action_callback_mapping
 
         callback_inputs: Union[List[State], Dict[str, State]]
+        # TODO-actions: Refactor the following lines to:
+        #     `callback_inputs = self.function.inputs + [State(*input.split(".")) for input in self.inputs]`
+        #     After refactoring that's mentioned above, test overwriting of the predefined action.
+        #     (by adding a new inputs/outputs to the overwritten action and check if it's working as expected)
         if self.inputs:
             callback_inputs = [State(*input.split(".")) for input in self.inputs]
+        elif hasattr(self.function, "inputs") and self.function.inputs:
```

Comment:
I guess we're going to have a few switches here while we have both the "old" function actions and the new class ones.

Let's have a consistent way of doing this everywhere to make it clearer. I think just `isinstance(function, CapturedActionCallable)` would work? No need to check if `self.function.inputs` is `False`y or not I think either.

Reply:
> I guess we're going to have a few switches here while we have both the "old" function actions and the new class ones. 

You're right, and after all actions become implemented as CapturedActionCallable, then the following line will be removed:
`else:
            callback_inputs = _get_action_callback_mapping(action_id=ModelID(str(self.id)), argument="inputs")`